sentry_sdk/scope.py

```diff
@@ -147,21 +186,183 @@ class Scope(object):
+            isolation_scope = Scope(ty="isolation")
+            _isolation_scope.set(isolation_scope)
+
+        return isolation_scope
+
+    @classmethod
+    def get_global_scope(cls):
+        # type: () -> Scope
+        """
+        Returns the global scope.
+
+        .. versionadded:: 1.XX.0
+        """
+        global _global_scope
+        if _global_scope is None:
+            _global_scope = Scope(ty="global")
+
+        return _global_scope
+
+    @classmethod
+    def _merge_scopes(cls, additional_scope=None, additional_scope_kwargs=None):
+        # type: (Optional[Scope], Optional[Dict[str, Any]]) -> Scope
+        """
+        Merges global, isolation and current scope into a new scope and
+        adds the given additional scope or additional scope kwargs to it.
+
+        .. versionadded:: 1.XX.0
+        """
+        if additional_scope and additional_scope_kwargs:
+            raise TypeError("cannot provide scope and kwargs")
+
+        global _global_scope
+
+        final_scope = copy(_global_scope) if _global_scope is not None else Scope()
+        final_scope._type = "merged"
+
+        isolation_scope = _isolation_scope.get()
+        if isolation_scope is not None:
+            final_scope.update_from_scope(isolation_scope)
+
+        current_scope = _current_scope.get()
+        if current_scope is not None:
+            final_scope.update_from_scope(current_scope)
+
+        if additional_scope is not None:
+            if callable(additional_scope):
+                additional_scope(final_scope)
+            else:
+                final_scope.update_from_scope(additional_scope)
+
+        elif additional_scope_kwargs:
+            final_scope.update_from_kwargs(**additional_scope_kwargs)
+
+        return final_scope
+
+    @classmethod
+    def get_client(cls):
+        # type: () -> Union[sentry_sdk.Client, sentry_sdk.client.NoopClient]
+        """
+        Returns the currently used :py:class:`sentry_sdk.Client`.
+        This checks the current scope, the isolation scope and the global scope for a client.
+        If no client is available a :py:class:`sentry_sdk.client.NoopClient` is returned.
+
+        .. versionadded:: 1.XX.0
+        """
+        current_scope = _current_scope.get()
+        if current_scope is not None and current_scope.client.is_active():
+            return current_scope.client
+
+        isolation_scope = _isolation_scope.get()
+        if isolation_scope and isolation_scope.client.is_active():
+            return isolation_scope.client
+
+        global _global_scope
+        if _global_scope is not None:
+            return _global_scope.client
+
+        return NoopClient()
+
+    def set_client(self, client=None):
+        # type: (Optional[sentry_sdk.Client]) -> None
+        """
+        Sets the client for this scope.
+        :param client: The client to use in this scope.
+            If `None` the client of the scope will be replaced by a :py:class:`sentry_sdk.NoopClient`.
```

Comment:
Does it ever make sense to call `set_client` without an actual client?

Reply:
This is for "deleteing" a client from the scope. This is more of an advanced use case, but maybe a person adds a custom client to the current scope that he/she later does not need anymore. Checked also with Francesco and he also things it makes sense, but will be rarely used.